packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -74,123 +90,522 @@ const propTypes = {
    */
   required: PropTypes.bool,
   /**
-   * @private Internal prop for showing date picker.
+   * @private
+   * Internal prop for showing date picker.
    */
   shouldShowPicker: PropTypes.bool,
   /**
    * The selected or entered date value to display in the date input.
    */
   value: PropTypes.string,
-  /**
-  * String that labels the current element. 'aria-label' must be present,
-  * for accessibility.
-  */
-  ariaLabel: PropTypes.string,
 };
 
 const defaultProps = {
+  ariaLabel: undefined,
   buttonRefCallback: undefined,
   inputAttributes: undefined,
+  inputRefCallback: undefined,
   isIncomplete: false,
   isInvalid: false,
   name: undefined,
   onBlur: undefined,
+  onButtonFocus: undefined,
   onChange: undefined,
   onClick: undefined,
   onFocus: undefined,
-  onButtonFocus: undefined,
   onKeyDown: undefined,
   required: false,
   placeholder: undefined,
   value: undefined,
-  ariaLabel: undefined,
 };
 
-class DatePickerInput extends React.Component {
-  constructor(props) {
-    super(props);
+const dateReducer = (state, inputValue) => ({
+  ...state,
+  ...inputValue,
+});
+
+const dateState = { day: '', month: '', year: '' };
+
+const DatePickerInput = (props) => {
+  const {
+    ariaLabel,
+    buttonRefCallback,
+    inputAttributes,
+    inputRefCallback,
+    intl,
+    isIncomplete,
+    isInvalid,
+    name,
+    onBlur,
+    onButtonFocus,
+    onChange,
+    onClick,
+    onFocus,
+    onKeyDown,
+    placeholder,
+    required,
+    value,
+    ...customProps
+  } = props;
 
-    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+  const [date, dateDispatch] = useReducer(dateReducer, dateState);
+  const [isFocused, setFocused] = useState(false);
+  const [dayInitialFocused, setDayInitialFocused] = useState(false);
+  const [monthInitialFocused, setMonthInitialFocused] = useState(false);
+  const [yearInitialFocused, setYearInitialFocused] = useState(false);
+  const dayInputRef = useRef();
+  const monthInputRef = useRef();
+  const yearInputRef = useRef();
+
+  const id = customProps.id ? customProps.id : undefined;
+  if (customProps.id) {
+    delete customProps.id;
```

Comment:
My vote is to make id a first class prop & we don't need to do this logic check.

Reply:
I too agree with making `id` as a first class prop.
Updated here: https://github.com/cerner/terra-framework/pull/1208/commits/ba9d3c9b977ded21c1db0789cbe2b1e43dd3c7c6